### PR TITLE
Cloudbroker scenarios

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+import random, unittest
+#from framework.api.client import Client
+
+
+
+class Test(TestcasesBase):
+
+    def setUp(self):
+        super().setUp()
+        self.user="{}@itsyouonline".format(self.whoami)
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: Ahmed will create a general method in the framework file which will create an account and return the accound id. Use it

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    def setUp(self):
+        super().setUp()
+        self.user="{}@itsyouonline".format(self.whoami) 
+        self.account,self.response=self.api.cloudbroker.accounts.create(self.user)
+        self.assertEqual(self.response.status_code, 200)
+        self.CLEANUP['accounts'].append(self.response.json())
+
+    def tearDown(self):
+        super().tearDown()
+        for accountId in self.CLEANUP['accounts']:
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: add it to the testbase tear down

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            self.api.cloudbroker.accounts.delete(accountId)
+        
+        for user in self.CLEANUP['users']:
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: add it to the testbase tear down

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        #. If accesstype read [R], should succeed to get account[C1] with user[U1]
+                                 , should fail to update account[C1] name or create cloudspace on it.
+        #. If accesstype write [RCX], should succeed to get account[C1] and create cloudspace on it
+                                    , should fail to updare account[C1] name. 
+        #. If accesstype admin [ARCXDU], should succeed to get , update [C1] name and  create cloudspace on it. 
+        """
+
+        self.lg.info("Create user [u1].")
+        user_data,response = self.api.cloudbroker.user.create(groups=["user"])
+        self.CLEANUP['users'].append(user_data["username"])
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: add to trear down

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        **Test Scenario:**
+   
+        #. Create user [u1].
+        #. Create account[C1] for main user and get this account  with main user,should succeed.
+        #. Add user[U1] to [C1]with access[accesstype], should succeed.
+        #. If accesstype read [R], should succeed to get account[C1] with user[U1]
+                                 , should fail to update account[C1] name or create cloudspace on it.
+        #. If accesstype write [RCX], should succeed to get account[C1] and create cloudspace on it
+                                    , should fail to updare account[C1] name. 
+        #. If accesstype admin [ARCXDU], should succeed to get , update [C1] name and  create cloudspace on it.
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: Add 'Delete' too

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        self.assertEqual(response.status_code, 404)
+
+        response= self.api.cloudbroker.accounts.deleteAccounts([accounts[2],accounts[3]])
+        self.assertEqual(response.status_code, 200)
+
+        for accountID in accounts:
+            response = self.api.cloudapi.accounts.get(accountID)
+            self.assertEqual(response.json()["status"],"DESTROYED")
+       
+    def test006_account_disable_and_enable(self):
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: split different test cases in separate methods

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        self.assertEqual(response.status_code,200)
+
+    @unittest.skip("https://github.com/0-complexity/openvcloud/issues/1355")
+    def test007_Update_account(self):
+        """ OVC-000
+        *Test case for Update account .*
+
+        **Test Scenario:**       
+        #. Update name of non-exist account [C1],should fail.
+        #. Update deleted account. should fail.
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: split

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        data,response = self.user_api.cloudapi.cloudspaces.create(accountId=c1_id, location=location, access=self.user)
+        self.assertEqual(response.status_code,200)
+
+    @unittest.skip("https://github.com/0-complexity/openvcloud/issues/1355")
+    def test007_Update_account(self):
+        """ OVC-000
+        *Test case for Update account .*
+
+        **Test Scenario:**       
+        #. Update name of non-exist account [C1],should fail.
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: split

File: functional_testing/Openvcloud/RESTful/testcases/testcase_base.py [hunk 9]
```diff
@@ -5,6 +5,10 @@
 from testconfig import config
 from framework.api.client import Client
 from framework.utils.utils import Utils
+from testconfig import config
+
+client_id = config['main']['client_id']
```
Comment (islamTaha12) [inline functional_testing/Openvcloud/RESTful/testcases/testcase_base.py]: use it inside the setupClass method

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py (hunk 5 above)
Comment (Dinaamagdy) [reply functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: there is no delete in cloudapi , and i need user to be in user group only

File: functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py (hunk 6 above)
Comment (Dinaamagdy) [reply functional_testing/Openvcloud/RESTful/testcases/cloudbroker/test01_accounts.py]: done

**Pull request closed without merging**